### PR TITLE
[MIG] controller_report_xls: Migration to 11.0

### PR DESCRIPTION
**Pull request by goliveirab**

### CODE REVIEW

**Comment by goliveirab** [comment]
DUMMY: https://git.vauxoo.com/vauxoo/quebec/-/merge_requests/84

File: controller_report_xls/reports/report_xls.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import base64
+import logging
+from io import BytesIO
+import lxml.html
+import xlwt
+from odoo import api, models
+
+from ..reports.xfstyle import css2excel
+
+# import simplejson
```
Comment (luisg123v) [inline controller_report_xls/reports/report_xls.py]: Remove commented code

File: controller_report_xls/__manifest__.py [hunk 2]
```diff
@@ -21,13 +21,13 @@
 ###############################################################################
```
Comment (luisg123v) [inline controller_report_xls/__manifest__.py]: Could you run the following command to remove all these old Python headers?

`find . -name "*.py" -type f -exec sed -i '/^[^#]/,$!d; /./,$!d' '{}' \;`

``

File: controller_report_xls/reports/report_xls.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+def get_css_style(csstext, style):
```
Comment (luisg123v) [inline controller_report_xls/reports/report_xls.py:14]: Is all this required?

Check how @WR-96 deals wit excel files using only Odoo requirements [here](https://git.vauxoo.com/vauxoo/apps/-/merge_requests/161).

On the other hand, if I'm not mistaking, excel is not strictly required, a CCSV would also do the trick (again, if I remember correctly).

**Review by luisg123v — CHANGES_REQUESTED** [review]
Please always squash.

File: controller_report_xls/reports/report_xls.py (hunk 3 above)
Comment (hugho-ad) [reply controller_report_xls/reports/report_xls.py:14]: @luisg123v 

Yes, but this module already does the job, there is not need to code again, 

We agree at https://www.vauxoo.com/web?debug=1#id=29976&action=470&model=project.task&view_type=form&menu_id=317

that is less expensive in time use this module just migrating it than develop the csv/xsls implementation.

Comment (hugho-ad) [reply controller_report_xls/reports/report_xls.py:14]: and since is just for Lodi, I don't see any need of developing new improvements

Comment (luisg123v) [reply controller_report_xls/reports/report_xls.py:14]: @hugho-ad thanks for clarifying, I agree, I though this was new code. Resolving.

**Comment by hugho-ad** [comment]
@goliveirab this is ready?

Stefan already approved it :+1:

**Review by luisg123v — APPROVED** [review]
LGTM, just a small comment regarding commits:

You have one commit touching two modules and performing two different actions:
- One to migrate the module `controller_report_xls`
- Another to use it in the module `product_pricelist_report_qweb`

Could you split it into two commits, please?

In addition, remember to leave only useful information in commit messages, e.g.:
- Remove `[FIX] EOF`
- This `[FIX] make it installable` could be an item in the migration list

**Comment by hugho-ad** [comment]
@goliveirab @luisg123v 

Stefan approve it

@goliveirab it's ready?


LGTM btw

**Comment by goliveirab** [comment]
> @goliveirab @luisg123v
> 
> Stefan approve it
> 
> @goliveirab it's ready?
> 
> LGTM btw

Yes!

**Comment by hugho-ad** [comment]
@goliveirab 

Just missing to attend Luis's comments about commit messages, could you please do it?

regards!

**Comment by goliveirab** [comment]
@hugho-ad   @luisg123v 

I just split the commit.... could you please check this?

Regards.

**Review by luisg123v — APPROVED** [review]
LGTM :+1:

CC @belisac

**Comment by luisg123v** [comment]
@goliveirab 
Note for future changes: ideally, the commit migrating the module should've gone first, as the other one depends on it.

**Pull request merged**